api/v1alpha1/ingressnodefirewall_webhook.go

```diff
@@ -204,21 +145,58 @@ func validateRule(rule IngressNodeFirewallProtocolRule, infRulesIndex, ruleIndex
 				infName, fmt.Sprintf("must be a valid %s rule: %s", rule.Protocol, reason))
 		}
 
-		if isConflict, reason := isConflictWithPinHoles(rule); isConflict {
+		if isConflict, reason, err := isConflictWithAllowRulesTransport(rule); err != nil {
+			return field.Invalid(field.NewPath("spec").Child("ingress").Index(infRulesIndex).Key("rules").Index(ruleIndex),
+				infName, fmt.Sprintf("must be a valid %s rule: %v", rule.Protocol, err))
+		} else if isConflict {
 			return field.Forbidden(field.NewPath("spec").Child("ingress").Index(infRulesIndex).Key("rules").Index(ruleIndex),
 				reason)
 		}
 	}
 	return nil
 }
 
-func isConflictWithPinHoles(rule IngressNodeFirewallProtocolRule) (bool, string) {
-	for _, pinhole := range pinholes[rule.Protocol] {
-		if isConflict, err := pinhole.isRuleConflict(rule); isConflict {
-			return true, err.Error()
+func isConflictWithAllowRulesTransport(rule IngressNodeFirewallProtocolRule) (bool, string, error) {
+	var allowRules []failsaferules.TransportProtoFailSafeRule
+	var err error
+	var isConflict bool
+	var start, end uint16
+
+	if rule.Protocol == ProtocolTypeTCP {
+		allowRules = failsaferules.GetTCP()
+	} else if rule.Protocol == ProtocolTypeUDP {
+		allowRules = failsaferules.GetUDP()
+	} else {
+		return false, "", fmt.Errorf("unable to determine conflict rules for unknown protocol: %q", rule.Protocol)
+	}
+
+	for _, allowRule := range allowRules {
+		if rule.ProtocolRule == nil {
+			return false, "", fmt.Errorf("expected ports to be defined for transport protocol")
+		}
+		if rule.ProtocolRule.IsRange() {
+			start, end, err = rule.ProtocolRule.GetRange()
+			if err != nil {
+				return false, "", fmt.Errorf("failed to get rule ports range: %v", err)
+			}
+			if isConflict, err = allowRule.IsConflictRange(start, end); err != nil {
+				return false, "", err
+			} else if isConflict {
+				return true, fmt.Sprintf("port range is in conflict with access to %s", allowRule.GetServiceName()), nil
```

Comment:
why not return port range conflict as an error and not return string at all ?

Reply:
I want to distinguish between an error and a reason for a conflict.

Reply:
If you want me to squash it into one return arg, I will. Dont feel strongly.

Reply:
I just noticed its empty string in most cases and sometimes error comes as err or string I think will make things easier if we use one way to report validation error , agree ?

Reply:
ya. agreed.